src/main/java/ArraySortedBag.java

```diff
@@ -1,75 +1,74 @@
         bag = (T[]) new Comparable[initialCapacity];
         rear = 0;
     }
 
     /**
-     * Doubles the size of the bag array and copy the
-     * contents over.
-     */
-    @SuppressWarnings("unchecked")
-    private void expandCapacity() {
-        T[] newBag = (T[]) new Comparable[bag.length * 2];
-        for (int i = 0; i < bag.length; ++i) {
-            newBag[i] = bag[i];
-        }
-        bag = newBag;
-    }
-
-    /**
-     * Shifts elements in an array down (towards index 0) towards
-     * the starting index specified. This method assumes that
-     * the calling function manages the rear field.
+     * Shifts elements in an array down (towards index 0) to the starting index specified. The element at the starting
+     * index will be overwritten.
      *
-     * @param start Index of where shift starts/stops and element
-     *              overwritten.
+     * @param start Index of element to be overwritten and where shifting moves down to.
      */
     private void shiftLeft(int start) {
-        for (int i = start; i < rear - 1; ++i) {
+        for (int i = start; i < rear - 1; i++) {
             bag[i] = bag[i + 1];
         }
         bag[rear - 1] = null;
     }
 
     /**
-     * Shifts elements in an array up (towards index rear) away
-     * from the starting index specified. This method assumes that
-     * the calling function manages the rear field and any
-     * need of expandCapacity().
+     * Shifts elements in an array up (towards index rear) away from the starting index specified. The array location
+     * at the specified starting index will be open. This method assumes there is room in the array to facilitate the
+     * shifting.
      *
-     * @param start Index of where shifting starts/stops and where
-     *              space is opened.
+     * @param start Index of where the array has a new open location and where shifting moves up from.
      */
     private void shiftRight(int start) {
-        for (int i = rear; i > start; --i) {
+        for (int i = rear; i > start; i--) {
             bag[i] = bag[i - 1];
         }
         bag[start] = null;
     }
 
     /**
-     * Find the index of a specified target element. If the element
-     * does not exist, return NOT_FOUND
+     * Find and return the index of a given target element within the collection. If no such element exists within the
+     * collection, a sentinel value of -1 (NOT_FOUND constant) is returned.
      *
-     * @param target Element to be found
-     * @return Index of the element if found, NOT_FOUND otherwise
+     * @param target Element to find the index of.
+     * @return Index of the target element within the collection, or -1 (NOT_FOUND constant) if no such element exists.
      */
-    private int sentinelIndexOf(T target) {
+    private int find(T target) {
         int searchIndex = 0;
         for (T bagElement : this) {
             if (bagElement.equals(target)) {
```

Comment:
It's typically more sane to check equality via the comparator in these types of cases. However, we know `T` is comparable, and `compareTo` mentions it should be consistent with `equals`. Sooooo. Maybe it's not worth getting into for first year.